src/order.ts

```diff
@@ -1,643 +1,230 @@
+  requestedDestination?: RequestedDestination;
   destinationLinkName?: string;
   releaseTime?: string;
   stopPrice?: number;
-  stopPriceLinkBasis?:
-    | 'MANUAL'
-    | 'BASE'
-    | 'TRIGGER'
-    | 'LAST'
-    | 'BID'
-    | 'ASK'
-    | 'ASK_BID'
-    | 'MARK'
-    | 'AVERAGE';
-  stopPriceLinkType?: 'VALUE' | 'PERCENT' | 'TICK';
+  stopPriceLinkBasis?: PriceLinkBasis;
+  stopPriceLinkType?: PriceLinkType;
   stopPriceOffset?: number;
-  stopType?: 'STANDARD' | 'BID' | 'ASK' | 'LAST' | 'MARK';
-  priceLinkBasis?:
-    | 'MANUAL'
-    | 'BASE'
-    | 'TRIGGER'
-    | 'LAST'
-    | 'BID'
-    | 'ASK'
-    | 'ASK_BID'
-    | 'MARK'
-    | 'AVERAGE';
-  priceLinkType?: 'VALUE' | 'PERCENT' | 'TICK';
+  stopType?: StopType;
+  priceLinkBasis?: PriceLinkBasis;
+  priceLinkType?: PriceLinkType;
   price?: number;
-  taxLotMethod?:
-    | 'FIFO'
-    | 'LIFO'
-    | 'HIGH_COST'
-    | 'LOW_COST'
-    | 'AVERAGE_COST'
-    | 'SPECIFIC_LOT';
+  taxLotMethod?: TaxLotMethod;
   orderLegCollection?: {
-    orderLegType?:
-      | 'EQUITY'
-      | 'OPTION'
-      | 'INDEX'
-      | 'MUTUAL_FUND'
-      | 'CASH_EQUIVALENT'
-      | 'FIXED_INCOME'
-      | 'CURRENCY';
+    orderLegType?: AssetType;
     legId?: number;
     instrument?: {
-      assetType?:
-        | 'EQUITY'
-        | 'OPTION'
-        | 'INDEX'
-        | 'MUTUAL_FUND'
-        | 'CASH_EQUIVALENT'
-        | 'FIXED_INCOME'
-        | 'CURRENCY';
+      assetType?: AssetType;
       cusip?: string;
       symbol?: string;
       description?: string;
     };
-    instruction?:
-      | 'BUY'
-      | 'SELL'
-      | 'BUY_TO_COVER'
-      | 'SELL_SHORT'
-      | 'BUY_TO_OPEN'
-      | 'BUY_TO_CLOSE'
-      | 'SELL_TO_OPEN'
-      | 'SELL_TO_CLOSE'
-      | 'EXCHANGE';
-    positionEffect?: 'OPENING' | 'CLOSING' | 'AUTOMATIC';
+    instruction?: Instruction;
+    positionEffect?: PositionEffect;
     quantity?: number;
-    quantityType?: 'ALL_SHARES' | 'DOLLARS' | 'SHARES';
+    quantityType?: QuantityType;
   }[];
   activationPrice?: number;
-  specialInstruction?:
-    | 'ALL_OR_NONE'
-    | 'DO_NOT_REDUCE'
-    | 'ALL_OR_NONE_DO_NOT_REDUCE';
-  orderStrategyType?: 'SINGLE' | 'OCO' | 'TRIGGER';
+  specialInstruction?: SpecialInstruction;
   orderId?: number;
   cancelable?: boolean;
   editable?: boolean;
-  status?:
-    | 'AWAITING_PARENT_ORDER'
-    | 'AWAITING_CONDITION'
-    | 'AWAITING_MANUAL_REVIEW'
-    | 'ACCEPTED'
-    | 'AWAITING_UR_OUT'
-    | 'PENDING_ACTIVATION'
-    | 'QUEUED'
-    | 'WORKING'
-    | 'REJECTED'
-    | 'PENDING_CANCEL'
-    | 'CANCELED'
-    | 'PENDING_REPLACE'
-    | 'REPLACED'
-    | 'FILLED'
-    | 'EXPIRED';
+  status?: OrderStatus;
   enteredTime?: string;
   closeTime?: string;
   tag?: string;
   accountId?: number;
   orderActivityCollection?: {
-    activityType?: 'EXECUTION' | 'ORDER_ACTION';
+    activityType?: ActivityType;
   }[];
-  replacingOrderCollection?: any[];
-  childOrderStrategies?: any[];
+  replacingOrderCollection?: Order[];
```

Comment:
Eyeballing this seemed to show I could re-use `Order` for `replacingOrderCollection` and `childOrderStrategies`. Is that right?